ui/src/shared/components/LayoutRenderer.js

```diff
@@ -33,20 +33,37 @@ class LayoutRenderer extends Component {
     if (!this.props.onPositionChange) {
       return
     }
+
+    let changed = false
+
     const newCells = this.props.cells.map(cell => {
       const l = layout.find(ly => ly.i === cell.i)
+
+      if (
+        cell.x !== l.x ||
+        cell.y !== l.y ||
+        cell.h !== l.h ||
+        cell.w !== l.w
+      ) {
+        changed = true
+      }
+
       const newLayout = {
         x: l.x,
         y: l.y,
         h: l.h,
         w: l.w,
       }
+
       return {
         ...cell,
         ...newLayout,
       }
     })
-    this.props.onPositionChange(newCells)
+
+    if (changed) {
```

Comment:
This seems like an optimization at a glance -- what does this do exactly?

Reply:
There is a call to onPositionChange when LayoutGrid is first instantiated, causing a put req. to dashboard, which was unnecessary because it was called with the same, unchanged init cell layout. So- we're checking if there have been any changes to cell layout to prevent the un-needed put.